### ISSUE TITLE
EarlyStopping raises TypeError when training with `val=False` setting, as BaseTrainer.fitness is None

### ISSUE THREAD

**Issue by jarombouts** [ultralytics/ultralytics#586]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

When training a segmentation model on my own custom dataset, I've found that a train run with setting `val=False` raises a TypeError after one epoch of training. Please see traceback:

```
Starting training for 2 epochs...
      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
        1/2         0G      2.769      4.483      4.415      1.707         32        256: 100%|██████████| 35/35 [07:29<00:00, 12.85s/it]
Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/pydevd.py", line 1499, in _exec
    func()
  File "/Users/jeroen/.pyenv/versions/<snip>/lib/python3.9/site-packages/ultralytics/yolo/cfg/__init__.py", line 237, in entrypoint
    func(cfg)
  File "/Users/jeroen/.pyenv/versions/<snip>/lib/python3.9/site-packages/ultralytics/yolo/v8/segment/train.py", line 151, in train
    model.train(**vars(cfg))
  File "/Users/jeroen/.pyenv/versions/<snip>/lib/python3.9/site-packages/ultralytics/yolo/engine/model.py", line 201, in train
    self.trainer.train()
  File "/Users/jeroen/.pyenv/versions/<snip>/lib/python3.9/site-packages/ultralytics/yolo/engine/trainer.py", line 182, in train
    self._do_train(int(os.getenv("RANK", -1)), world_size)
  File "/Users/jeroen/.pyenv/versions/<snip>/lib/python3.9/site-packages/ultralytics/yolo/engine/trainer.py", line 343, in _do_train
    self.stop = self.stopper(epoch + 1, self.fitness)
  File "/Users/jeroen/.pyenv/versions/<snip>/lib/python3.9/site-packages/ultralytics/yolo/utils/torch_utils.py", line 387, in __call__
    if fitness >= self.best_fitness:  # >= 0 to allow for early zero-fitness stage of training
TypeError: '>=' not supported between instances of 'NoneType' and 'float'
``` 

### Environment

- YOLO V8, `ultralytics` Pip package `8.0.17`
- OS: MacOS 12.6.2 on 2020 Macbook Pro 13''
- CPU/GPU: M1 

### Minimal Reproducible Example

```bash
HYDRA_FULL_ERROR=1 yolo task=segment mode=train model=yolov8n-seg.pt \
                        data="$DATA" epochs=2 batch=64 \
                        imgsz=240 pretrained=True val=False \
                        cache=false lr=0.005
```

### Additional

Can be fixed by changing line 344 in `ultralytics.yolo.engine.trainer`

```python
    self.stop = self.stopper(epoch + 1, self.fitness) if self.fitness is not None else False
```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by AyushExel** [comment]
Thanks for the report. Yes please submit a PR if you'd like to and tag me

**Comment by glenn-jocher** [comment]
@jarombouts closing as merged in #589. Please let us know if you have any other problems!
